src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java

```diff
@@ -2902,6 +2920,44 @@ private void fireLineWidgetsChanged()
       AceEditor.this.fireEvent(new LineWidgetsChangedEvent());
    }
    
+   private static class ScopeTimer
+   {
+      public ScopeTimer(final AceEditor editor)
+      {
+         timer_ = new Timer()
+         {
+            @Override
+            public void run()
+            {
+               if (row_ >= editor.getRowCount())
+                  return;
+               
+               row_ += 200;
```

Comment:
@jjallaire I think you're right; on my MBP it seems to take from 1~2ms to tokenize a single line so this could potentially tip that bounds.